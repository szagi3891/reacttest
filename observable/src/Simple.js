// @flow

import React, { Component } from 'react';

type PropsType = {|
    title?: string,
    onClick?: (imageTitle: string) => void,
|};

class Simple extends Component {

    props: PropsType;

    constructor(props: PropsType) {
        super(props);
        console.warn('jakis konstruktor');
    }

    componentWillMount() {
        console.warn('component will mount');
    }

    componentDidMount() {
        console.warn('component did mount');
    }

/*
    componentWillReceiveProps(nextProps: PropsTypeIn) {
    }

    componentWillUnmount() {
    }
*/

    render(): React.Element<*> {
        const { title } = this.props;

        return (
            <div className="SimpleWrapper">
                { title }
                <button onClick={this._onClick.bind(this)}>kliknij</button>
            </div>
        );
    }

    _onClick() {
        const { onClick } = this.props;

        if (onClick) {
            onClick(window.ImageConst);
        }
    }
}

export default Simple;
