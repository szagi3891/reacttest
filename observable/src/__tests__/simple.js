// @flow

import React from 'react';
import { mount } from 'enzyme';
import sinon from 'sinon';
import Simple from '../Simple';

describe('simple', () => {

    it('podstawowy', () => {

      sinon.spy(Simple.prototype, 'componentDidMount');
      const wrapper = mount(<Simple />);
      expect(Simple.prototype.componentDidMount.calledOnce).toEqual(true);
    });

    it('podstawowy2', () => {
        window.ImageConst = 'Jakis tajny tekst';

        const wrapper = mount(<Simple title="baz" />);
        expect(wrapper.props().title).toEqual("baz");
        wrapper.setProps({title: "foo"});
        expect(wrapper.props().title).toEqual("foo");


        const onClick = sinon.spy();
        wrapper.setProps({onClick: onClick});
        wrapper.find('button').simulate('click');
        expect(onClick.calledOnce).toEqual(true);


        let messRet = null;
        wrapper.setProps({onClick: (label: string) => {
            messRet = label;
        }});
        wrapper.find('button').simulate('click');
        expect(messRet).toEqual("Jakis tajny tekst");

        //http://airbnb.io/enzyme/docs/api/mount.html
    });
});
