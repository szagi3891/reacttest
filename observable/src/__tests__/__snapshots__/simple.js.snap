exports[`simple snapshot 1`] = `
<div
  className="SimpleWrapper">
  baz
  <button
    onClick={[Function]}>
    kliknij
  </button>
</div>
`;
